### PR TITLE
feat: use showSnackbarMessage in viewmodel

### PR DESCRIPTION
**Pull request by williamwue**
Instead of show snackbar in fragment directly. Prepare for do remove garden planting action later (control delete action and show message action both in viewmodel).

### CODE REVIEW

File: app/src/main/java/com/google/samples/apps/sunflower/utilities/SingleLiveEvent.kt [hunk 1]
```diff
@@ -0,0 +1,76 @@
+/*
```
Comment (XinyueZ) [inline app/src/main/java/com/google/samples/apps/sunflower/utilities/SingleLiveEvent.kt]: Use `EventObserver` pattern instead `SingleLiveEvent`
Check: http://tinyurl.com/y7e8hq4v
There's a blog link which describes how different between `EventObserver` & `SingleLiveEvent` which is important.

File: app/src/main/java/com/google/samples/apps/sunflower/PlantDetailFragment.kt [hunk 2]
```diff
@@ -60,10 +62,16 @@ class PlantDetailFragment : Fragment() {
             setLifecycleOwner(this@PlantDetailFragment)
             fab.setOnClickListener { view ->
```
Comment (XinyueZ) [inline app/src/main/java/com/google/samples/apps/sunflower/PlantDetailFragment.kt]: `view` which is local will never be used, use  `_` instead.

File: app/src/main/java/com/google/samples/apps/sunflower/PlantDetailFragment.kt [hunk 3]
```diff
@@ -60,10 +62,16 @@ class PlantDetailFragment : Fragment() {
...
                 plantDetailViewModel.addPlantToGarden()
-                Snackbar.make(view, R.string.added_plant_to_garden, Snackbar.LENGTH_LONG).show()
+                plantDetailViewModel.showSnackbarMessage(R.string.added_plant_to_garden)
```
Comment (XinyueZ) [inline app/src/main/java/com/google/samples/apps/sunflower/PlantDetailFragment.kt]: The Line 64-65 should happen asynchronously, that means the `message` would be shown just after `adding` behavior being caused which is incorrect information while the `inserting` is still running underway possibly.

**My opinion**  here is to use the **return** value(`LiveData<Long>`) of insert DAO which can be used by the view-model and show  `message` asynchronously.

File: app/src/main/java/com/google/samples/apps/sunflower/utilities/SingleLiveEvent.kt (hunk 1 above)
Comment (williamwue) [reply app/src/main/java/com/google/samples/apps/sunflower/utilities/SingleLiveEvent.kt]: @XinyueZ OK, I'll learn it.

File: app/src/main/java/com/google/samples/apps/sunflower/PlantDetailFragment.kt (hunk 2 above)
Comment (williamwue) [reply app/src/main/java/com/google/samples/apps/sunflower/PlantDetailFragment.kt]: Got it.

**Comment by mlykotom** [comment]
Thank you folks for the contribution, I'm going to close this as obsolete given the screen has been rewritten.

**Pull request closed without merging**